### PR TITLE
python3Packages.monitorcontrol: init at 3.1.0

### PR DESCRIPTION
**Pull request by newAM**
## Description of changes

Repository: https://github.com/newAM/monitorcontrol

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/monitorcontrol/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+, poetry-core
+, pyudev
+, pytestCheckHook
+, voluptuous
+}:
+
+buildPythonPackage rec {
+  pname = "monitorcontrol";
+  version = "3.1.0";
+  format = "pyproject";
```
Comment (natsukium) [inline pkgs/development/python-modules/monitorcontrol/default.nix:14]:
```suggestion
  pyproject = true;
```
Could you please use `pyproject = true;` instead?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters

File: pkgs/development/python-modules/monitorcontrol/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+, poetry-core
+, pyudev
+, pytestCheckHook
+, voluptuous
+}:
+
+buildPythonPackage rec {
```
Comment (natsukium) [inline pkgs/development/python-modules/monitorcontrol/default.nix:11]: Is this a library? If it is an application and not intended to be used as a library, please use `buildPythonApplication` and move it to pkgs/by-name or somewhere.

File: pkgs/development/python-modules/monitorcontrol/default.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    hash = "sha256-fu0Lm7Tcw7TCCBDXTTY20JBAM7oeesyeHQFFILeZxX0=";
+  };
+
+  nativeBuildInputs = [
+    poetry-core
+  ];
+
+  propagatedBuildInputs = [
+    pyudev
+    voluptuous
```
Comment (natsukium) [inline pkgs/development/python-modules/monitorcontrol/default.nix:31]:
```suggestion
```
`voluptuous` is a test dependency, isn't it?

File: pkgs/development/python-modules/monitorcontrol/default.nix [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
+  checkInputs = [
+    voluptuous
```
Comment (natsukium) [inline pkgs/development/python-modules/monitorcontrol/default.nix:39]: The test dependency should go to nativeCheckInputs.

File: pkgs/development/python-modules/monitorcontrol/default.nix (hunk 2 above)
Comment (newAM) [reply pkgs/development/python-modules/monitorcontrol/default.nix:11]: It's primarily a library with a CLI for testing purposes.

**Comment by newAM** [comment]
Thanks for the review @natsukium!  I applied you changes.

File: pkgs/development/python-modules/monitorcontrol/default.nix (hunk 2 above)
Comment (natsukium) [reply pkgs/development/python-modules/monitorcontrol/default.nix:11]: I see. Thanks for clarifying.

**Comment by natsukium** [comment]
@ofborg build python310Packages.monitorcontrol python311Packages.monitorcontrol

**Review by natsukium — APPROVED** [review]
LGTM, thanks.

**Pull request merged**